src/fmu/dataio/_fmu_provider.py

```diff
@@ -1,292 +1,387 @@
 """Module for DataIO _FmuProvider
 
-The FmuProvider will from the run environment and eventually API's detect valid
+The FmuProvider will from the environment and eventually API's detect valid
 fields to the FMU data block.
 
-Note that FMU may potentially have different providers, e.g. ERT2 vs ERT3
+Note that FMU may potentially have different providers, e.g. ERT versions
 or it can detect that no providers are present (e.g. just ran from RMS interactive)
+
+Note that establishing the FMU case metadata for a run, is currently *not* done
+here; this is done by code in the InitializeCase class.
+
+From ERT v. 5 (?), the following env variables are provided in startup (example):
+
+_ERT_EXPERIMENT_ID:   6a8e1e0f-9315-46bb-9648-8de87151f4c7
+_ERT_ENSEMBLE_ID:   b027f225-c45d-477d-8f33-73695217ba14
+_ERT_SIMULATION_MODE:   test_run
+
+and during a Forward model:
+
+_ERT_EXPERIMENT_ID:   6a8e1e0f-9315-46bb-9648-8de87151f4c7
+_ERT_ENSEMBLE_ID:   b027f225-c45d-477d-8f33-73695217ba14
+_ERT_SIMULATION_MODE:   test_run
+_ERT_ITERATION_NUMBER:   0
+_ERT_REALIZATION_NUMBER:   0
+_ERT_RUNPATH:   /scratch/fmu/jriv/01_drogon_ahm/realization-0/iter-0/
+
 """
 from __future__ import annotations
 
 import json
-import pathlib
-import re
+import os
 from copy import deepcopy
 from dataclasses import dataclass, field
+from enum import Enum, auto
 from os import environ
 from pathlib import Path
-from typing import Any, Final, Optional
+from typing import Final, Optional
 from warnings import warn
 
 from fmu.config import utilities as ut
 
 from . import _utils
+from ._definitions import FmuContext
 from ._logging import null_logger
 
-# case metadata relative to rootpath
-ERT2_RELATIVE_CASE_METADATA_FILE: Final = "share/metadata/fmu_case.yml"
+# case metadata relative to casepath
+ERT_RELATIVE_CASE_METADATA_FILE: Final = "share/metadata/fmu_case.yml"
 RESTART_PATH_ENVNAME: Final = "RESTART_FROM_PATH"
 
 logger: Final = null_logger(__name__)
 
 
-def _get_folderlist(current: Path) -> list:
-    """Return a list of pure folder names incl. current casepath up to system root.
+class FmuEnv(Enum):
+    EXPERIMENT_ID = auto()
+    ENSEMBLE_ID = auto()
+    SIMULATION_MODE = auto()
+    REALIZATION_NUMBER = auto()
+    ITERATION_NUMBER = auto()
+    RUNPATH = auto()
```

Comment:
I think a pydantic model w/from os-env static method is the way to go.

Outline:
```python
from pydantic import BaseModel, Field
class FmuEnv(BaseModel):
    experiment_id: str = Field(alias="_ERT_EXPERIMENT_ID")
    ...

    @statistics
    def from_osenv():
        return FmuEnv.model_validate(os.os.environ)
```